app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -220,6 +220,13 @@ class Mage_Adminhtml_Block_Widget_Grid extends Mage_Adminhtml_Block_Widget
         ],
     ];
 
+    /**
+     * Allow edit status
+     *
+     * @var array<string, bool>|null
+     */
+    protected ?array $isAllowed = null;
```

Comment:
```suggestion
    protected array $isAllowed = [];
```

wouldn't it be simpler to always have an array? the same way as https://github.com/OpenMage/magento-lts/pull/3785?